### PR TITLE
appsso: add tls scenario guides

### PR DESCRIPTION
**Pull request by sarahwalther**
[#183456128]

Which other branches do you want a technical writer to cherry-pick this PR to (if any)?

It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

### CODE REVIEW

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 1]
```diff
@@ -0,0 +1,785 @@
+# Authserver TLS Scenario Guides
+
+`AuthServer` is a piece of security infrastructure. It is imperative to configure TLS for it, so that its _issuer
+URI_'s scheme is `https://`.
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:4]: **[for info, not blocking]**

OpenID Connect mandates TLS [spec](https://openid.net/specs/openid-connect-core-1_0.html#TLSRequirements)
OAuth 2 mandates TLS too [rfc](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-07#section-3.1)

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+
+`AuthServer.spec.tls` accommodates different scenarios for how to obtain a TLS certificate. Pick the scenario that
+matches your case.
+
+<p class="note">
+<strong>Note:</strong>
+The recommended path is to install AppSSO with a [default issuer](#i-have-a-default-issuer-_recommended_). In that case,
+you can omit `AuthServer.spec.tls` entirely and a TLS certificate is obtained automatically.
+</p>
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:13]: **[praise]** neat!

File: app-sso/service-operators/issuer-uri-and-tls.hbs.md [hunk 3]
```diff
@@ -104,6 +104,8 @@ spec:
 
 Confirm that your `AuthServer` serves a TLS certificate from the specified issuer by visiting its `{.status.issuerURI}`.
 
+> 👉 Learn more about [different TLS Scenarios](./tls-scenario-guides.hbs.md).
```
Comment (Kehrlann) [inline app-sso/service-operators/issuer-uri-and-tls.hbs.md:107]: **[question]** is that the required language?

The word "Scenarios" is not super clear to me. I'm thinking something along the lines of _Learn more about different ways to configure TLS for your AuthServer_ or something similar?
Edit: after reading the other page, maybe Scenario is the right word

This link should probably feature more prominently at the top of the page ; and with a bit more explanation "there are many use-cases with certificates, e.g. certs managed outside the cluster, wildcards, etc".

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 4]
```diff
@@ -0,0 +1,785 @@
...
+## I have a default issuer _(recommended)_
+
+> Using [a default issuer](../platform-operators/configuration.md#default_authserver_clusterissuer) is the recommended
+approach. It appeals best to the split the responsibilities of _platform operators_ and _service operators_.
+> In this case, the `Authserver.spec.tls` field can be omitted entirely.
+
+> If you need to verify whether `AppSSO` was installed with a default issuer, you can run
+> `kctrl package installed get --namespace tap-install --package-install tap --values-file-output tap-values.yaml`.
+> Then check your `tap-values.yaml` file. If there is a `shared.ingress_issuer`, you have a default issuer.
+> Prerequisite: Ensure [kctrl](https://carvel.dev/blog/kctrl-release-blog/) is installed.
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:73]: **[question]** Should this be in block-quotes?

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 5]
```diff
@@ -0,0 +1,785 @@
...
+      key.pem: $(privateKey)
+      pub.pem: $(publicKey)
+```
+
+## I have a `ClusterIssuer`
+
+A `ClusterIssuer` is a cluster-scoped API provided by [cert-manager](https://cert-manager.io) from which certificates
+can be obtained programmatically.
+
+For purposes of completeness and simplicity, puts all resources into a single YAML file and
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:121]:
```suggestion
For purposes of completeness and simplicity, this scenario puts all resources into a single YAML file and
```

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 6]
```diff
@@ -0,0 +1,785 @@
...
+uses [Let's Encrypt](https://letsencrypt.org/)'s production API. It is likely, however, that the `ClusterIssuer` is
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:122]: **[suggestion]**

Add 
> Learn more about [Let's Encrypt with cert-manager](https://cert-manager.io/docs/configuration/acme/)

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 7]
```diff
@@ -0,0 +1,785 @@
...
+A `Certificate` is a Namespace-scoped API provided by [cert-manager](https://cert-manager.io) which represents a TLS
+certificate obtained from a `(Cluster)Issuer`. To create `Certificate`, you need to know the name and kind of your
+issuer.
+
+These scenarios use [Let's Encrypt](https://letsencrypt.org/)'s production API and assume that a
+`ClusterIssuer` by the name `letsencrypt-production` exists.
+See [the `ClusterIssuer` scenario](#i-have-a-clusterissuer-_recommended_) for how to set the issuer up.
+
+When using `Certificate`, its `.spec.dnsNames` must contain the FQDN of the templated issuer URI. Make sure that your
+AppSSO package installation's `domain_name` and `domain_template` are compatible with your DNS name.
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:268]: **[suggestion]** Is it worth adding a "troubleshooting" section about what happens when the DNS dnsNames are incorrect?

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 8]
```diff
@@ -0,0 +1,785 @@
...
+  secretTemplate:
+    type: Opaque
+    stringData:
+      key.pem: $(privateKey)
+      pub.pem: $(publicKey)
+```
+
+## I have an existing TLS certificate
+
+If you have an existing TLS certificate and private key, then this scenario is yours.
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:426]: **[suggestion]** The tone is different from the other scenarios, make it more formal.
**[suggestion]** Add "For example, your certificate was created outside the cluster".
**[suggestion]** Refer to the "troubleshooting" section suggested in the above scenario.

File: app-sso/service-operators/tls-scenario-guides.hbs.md [hunk 9]
```diff
@@ -0,0 +1,785 @@
...
+stringData:
+  #! --- ReplaceMe - certificate and private key for "sso.login.my-tap.example.com ---
+  tls.crt: |
+    -----BEGIN CERTIFICATE-----
+    # redacted
+    -----END CERTIFICATE-----
+  tls.key: |
+    -----BEGIN PRIVATE KEY-----
+    # redacted
+    -----END PRIVATE KEY-----
```
Comment (Kehrlann) [inline app-sso/service-operators/tls-scenario-guides.hbs.md:458]: **[suggestion]** Spell out in the description of the scenario that the certificate must be stored in a `Secret`, with the keys `tls.crt` and `tls.key`

**Review by Kehrlann — APPROVED** [review]
Great work, a few suggestions but nothing blocking.

File: app-sso/service-operators/tls-scenario-guides.hbs.md (hunk 7 above)
Comment (sarahwalther) [reply app-sso/service-operators/tls-scenario-guides.hbs.md:268]: I like that idea. Maybe we can merge this for now and see if we have time before the end of the week to add a troubleshooting section?

File: app-sso/service-operators/tls-scenario-guides.hbs.md (hunk 9 above)
Comment (sarahwalther) [reply app-sso/service-operators/tls-scenario-guides.hbs.md:458]: You're suggesting adding a note to draw attention to the secret, right?

Comment (Kehrlann) [reply app-sso/service-operators/tls-scenario-guides.hbs.md:463]: Yes.

File: app-sso/service-operators/tls-scenario-guides.hbs.md (hunk 7 above)
Comment (Kehrlann) [reply app-sso/service-operators/tls-scenario-guides.hbs.md:269]: Yes, good idea!

**Comment by sarahwalther** [comment]
Ready for review, @YugangW . We will address the remaining suggestions in a separate PR.

**Comment by YugangW** [comment]
Thanks, Sarah. Merged.

**Pull request merged**